### PR TITLE
feat: Style video error screen

### PR DESCRIPTION
**Pull request by aronshamash**
- Use poster image when avaliable
- Retry button just refreshes the page

### Before

 
screen shot 2018-09-18 at 10 44 46

### After

screen shot 2018-09-18 at 10 45 02

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/07136266b09ae83ecb859fc3d6949d8183f96001/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/888c4e1f530574c75ed18c14a16310ca92e8da5d/index.html

File: packages/video/src/video-error.web.js [hunk 1]
```diff
@@ -1,21 +1,53 @@
...
+          title="Retry"
+        />
+      </View>
+    </View>
+  );
+};
 
 VideoError.propTypes = {
   height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  poster: PropTypes.shape({ uri: PropTypes.string.isRequired }),
```
Comment (jimhunty) [inline packages/video/src/video-error.web.js]: Is it likely other properties might be added?

File: packages/video/src/video-error.web.js [hunk 2]
```diff
@@ -1,21 +1,53 @@
...
   width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
 };
 
+VideoError.defaultProps = {
+  poster: ""
```
Comment (jimhunty) [inline packages/video/src/video-error.web.js]: Doesnt match the prop type

File: packages/video/src/video-error.web.js [hunk 3]
```diff
@@ -1,21 +1,53 @@
...
+    );
+  };
+
+  return (
+    <View style={[styles.background, { width, height }]}>
+      <View style={styles.posterContainer}>{renderPosterImage()}</View>
+      <View style={styles.textContainer}>
+        <Text style={styles.heading}>Something&apos;s gone wrong</Text>
+        <Text style={styles.body}>
+          Please check your network connection and retry to try again
```
Comment (jimhunty) [inline packages/video/src/video-error.web.js]: Sentence needs work.

File: packages/video/src/video-error.web.js [hunk 4]
```diff
@@ -1,21 +1,53 @@
...
+        </Text>
+        <Button
+          onPress={() => {
+            if (typeof window !== "undefined") {
```
Comment (jimhunty) [inline packages/video/src/video-error.web.js]: Should this wrap the button and we change the message based on this?

File: packages/video/src/video-error.web.js (hunk 1 above)
Comment (aronshamash) [reply packages/video/src/video-error.web.js]: Changed to just be a string. Originally was an object to be consistent with `video-prop-types.js`

File: packages/video/src/video-error.web.js (hunk 3 above)
Comment (aronshamash) [reply packages/video/src/video-error.web.js]: So this is a compromise situation. This screen could display if it is an network error or an invalid id which refreshing will not help. We would need to do more work to be able to differentiate between different types of error. We need some wording which makes sense either for most common errors or for all error types. Suggestions as to better copy?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/05e75c4b7751bde68884122b4eecbdefb3589447/index.html

File: packages/video/src/styles/index.web.js [hunk 5]
```diff
@@ -1,33 +1,69 @@
...
-    color: "white",
-    width: "auto",
-    height: "auto",
-    marginBottom: spacing(2)
+  textContainer: {
+    position: "absolute",
+    maxWidth: "80%"
+  },
+  posterContainer: {
+    filter: "blur(8px)",
```
Comment (craigbilner) [inline packages/video/src/styles/index.web.js]: :cake:

File: packages/video/src/video-error.web.js (hunk 3 above)
Comment (colincclark) [reply packages/video/src/video-error.web.js]: > Please check your network connection

with a retry button should suffice?

File: packages/button/src/styles/index.web.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ const styles = {
     alignItems: "center",
     backgroundColor: colours.functional.action,
     borderRadius: 2,
+    borderColor: colours.functional.action,
```
Comment (colincclark) [inline packages/button/src/styles/index.web.js]: alphabetty spaghetti

File: packages/video/src/video-error.web.js (hunk 2 above)
Comment (colincclark) [reply packages/video/src/video-error.web.js]: it seems to be an object though? https://github.com/newsuk/times-components/pull/1337/files#diff-7f35ad8aff234a89efbd14e42fd7171eR16

File: packages/video/src/video-error.web.js [hunk 7]
```diff
@@ -1,21 +1,58 @@
...
+      />
+    );
+  };
+
+  const renderRetryButton = () => {
+    if (typeof window === "undefined") return null;
+    return (
+      <Button
+        onPress={() => {
+          window.location.reload();
```
Comment (colincclark) [inline packages/video/src/video-error.web.js]: why doesn't this tie in with the API call and re-try that instead?

File: packages/video/src/video-error.web.js [hunk 8]
```diff
@@ -1,21 +1,58 @@
...
-const VideoError = ({ height, width }) => (
-  <View style={[styles.background, { width, height }]}>
-    <Text style={styles.heading}>Video unable to play</Text>
-    <Text style={styles.body}>
-      Please check your network connection and try refreshing the page. If that
-      doesn&apos;t work, please try again later
-    </Text>
-  </View>
-);
+const VideoError = ({ height, width, poster }) => {
```
Comment (colincclark) [inline packages/video/src/video-error.web.js]: there are no changes to the tests or the showcase that passes in this new `poster` prop

**Pull request closed without merging**